### ISSUE TITLE
SingulierPluriel by Giorgio Chiappa

### ISSUE THREAD

**Issue by valiantone** [TechLabs-Berlin/project_proposals#2]
### What is your project's name?
SingulierPluriel. A user-friendly learning management system geared towards language learning and teaching

### What's the problem you are trying to solve?
In my work as an educator and language teacher, I have used many different learning management systems (LMS), both in an academic context (LMS like Blackboard, Moodle, etc.) and in adult education (VHS-Cloud). A lot of them have a certain difficulty of use in common — they abound in redundant sections, visual clutter, and they are often cumbersome to navigate. Hence my desire to create an LMS that both addresses these issues and allows for the hosting and creating of interesting digital learning experiences to complement language classes.

### What's your idea for a solution to the given "problem"?
The user research (still in progress) I have conducted up to now amongst my colleagues and our students has highlighted the following problem areas, in which pain points tend to arise the most: - navigation/ease of use (LMS has too many sections whose function is not always clear, no back/forth function, file extension has to be specified before upload etc.); - transparency (e.g. students cannot find their grades or keep track of their assignments; group communication or communication with instructors via the LMS is too cumbersome); - personalization (e.g. both students and course instructors do not feel they can tailor their LMS experience to their needs and goals). My solutions to these problems could be summed up in the following verbs: simplify (for instance: keep only the most relevant sections and keywords which are all marked with clear and transparent language - e.g. “Course Materials”, “Communication”, “Grades”); centralise (allow the LMS to fulfill all or most student and course instructor needs without needing to resort to other tools); customise (allow students to create personal spaces with their notes and resources / create an agile experience for course instructors who will be able integrate a variety of tools and formats into the LMS ).

### Which tracks do you think could be involved?
DL/DS, WD

### What are the most important challenges & bottlenecks of your project?
Mainly the fact that my idea of an LMS also presupposes a certain facility in integrating many different tools and formats — how can I make sure that it works smoothly, especially if content or links created elsewhere come into play? Another tough nut to crack might be the creation of a functional communication system inside the LMS, allowing groups to cooperate well and ensuring smooth communication between course instructors and students. Easy to say, harder to pull off.

### How do you imagine Deep Learning and/or Data Science could contribute to this project?
For one, I was thinking that DL techies might help me envisage how course instructors could generate learning materials inside the LMS — for instance by creating exercises, texts, or providing AI-generated images to decorate or complement the tasks. Also, I was wondering whether we could use DL knowledge to improve navigation. Aside from clicking your way around the LMS, are there maybe other ways to get the job done — for instance by using a search engine, or even your voice? This could be particularly interesting for users with disabilities, but it would benefit everyone in terms of accessibility and ease of use.

### How do you imagine Web Development could contribute to this project?
Well, there’s no LMS if I have no one to translate my ideas into a web app! So of course WD is crucial for my project. I thinking that speaking to well-informed WD techies might also help me understand what is actually feasible and desirable in terms of how the LMS looks like and works.

### What are the most important success factors of your project?
Familiarising oneself with the users, knowing their pain points and needs, understanding how the LMS might be actually integrated into the field of language learning but also in a broader academic environment (for UX design — that would be my side of things!) ; being able to design an agile web app for the LMS that responds to the principles I have listed above, namely “simplify”, “centralise”, “customise” (for WD) ; allowing content generation inside the LMS and improving navigation with AI (for DS/DL ).

### Share your resources here:
[Figma page](https://www.figma.com/file/3gYNGRnV8JohRe1epgpCVU/25112023PersonasScenariosStoryboardsfirst-draft?type=whiteboard&node-id=0%3A1&t=ET1ge4BPkdQSsWpz-1) where I keep personas, scenarios, storyboards. This is still very provisional, and I have to get back to it now that I have conducted more user research.

**Comment by valiantone** [comment]
> **Which tracks do you think could be involved?**
> DL/DS, WD

You may have overlooked this since you are from the UX track yourself, but I imagine that you will need/want to work with other designer on this problem?

**Comment by valiantone** [comment]
In project phase, we will start to work on an MVP. Within ten weeks it will be only possible to create an early concept and some features which can then be demonstrated in your prototype in final project presentations.

Here's the answers, we need from your side:

- How can UX techies collaborate and ideate on this problem? What's the gameplan to roll this out?
- WebDev seems a little straight forward, you will need a simple website or web-app with some dummy lessons added to it that users can engage with
- **Other WebDev features that can be built include:** login/authentication, progress tracking, reminders etc.Anything else might be out of scope for WD techies

**Comment by valiantone** [comment]
On the DL/DS side of things, great ideas, but they may be out of the scope of feasibility for techies that are just beginning their journey in the realm of AI. Most of the tasks you described are ambitious and require some practical expertise. Here are some things that are still possible for our DS techies to do:

- Recommend similar courses to the currently designed course
- Help with full text search to find content within curriculum
- Predict popularity, duration, time to completion, completion probability etc. which might be beneficial tools for content creators

Would avoid any use of DL technologies or techies for this project as I cannot imagine any features that could be relevant to proposed application.

**Comment by vanniberna** [comment]
> > **Which tracks do you think could be involved?**
> > DL/DS, WD
> 
> You may have overlooked this since you are from the UX track yourself, but I imagine that you will need/want to work with other designer on this problem?

Oh yes, of course! I already have a study group with whom I am working on similar themes since they also want to tackle language learning and didactics in general. I will start by asking them, though of course I would gladly be open to include other designers as well.

**Comment by vanniberna** [comment]
> In project phase, we will start to work on an MVP. Within ten weeks it will be only possible to create an early concept and some features which can then be demonstrated in your prototype in final project presentations.
> 
> Here's the answers, we need from your side:
> 
>     * How can UX techies collaborate and ideate on this problem? What's the gameplan to roll this out?
> 
>     * WebDev seems a little straight forward, you will need a simple website or web-app with some dummy lessons added to it that users can engage with
> 
>     * **Other WebDev features that can be built include:** login/authentication, progress tracking, reminders etc.Anything else might be out of scope for WD techies

As far as other UX techies are concerned, I think that - aside from collegial feedback - I could definitely use some help in the UI design side of things. I am sure I will have more command of UI design once I have tackled the corresponding section in our Edyoucated materials, but I already know myself enough to say I am more a verbal than a visual person, so my designs might come out a bit rough around the edges at first! Coincidentally, one thing a lot of students said in my user research was that they find the design of their current LMS unappealing, so that is something I would like to get right in my project. 

Could you please clarify what you mean exactly by "gameplan"? Do you mean a possible work plan to design and complete the product with other UX techies?

Thanks for your comments re: scope and how WebDev techies might be involved — duly noted.

**Comment by vanniberna** [comment]
> On the DL/DS side of things, great ideas, but they may be out of the scope of feasibility for techies that are just beginning their journey in the realm of AI. Most of the tasks you described are ambitious and require some practical expertise. Here are some things that are still possible for our DS techies to do:
> 
>     * Recommend similar courses to the currently designed course
> 
>     * Help with full text search to find content within curriculum
> 
>     * Predict popularity, duration, time to completion, completion probability etc. which might be beneficial tools for content creators
> 
> 
> Would avoid any use of DL technologies or techies for this project as I cannot imagine any features that could be relevant to proposed application.

Thank you for clarifying this. The functions you propose are already more than enough, especially the full-text search function. I was wondering if the latter could be implemented more holistically for the LMS, e.g. having a search function that appears on your home page after log-in and can allow you to be taken immediately to the section you're looking for (for instance: you log in, and instead of having to click on "Courses" and then "Italian A1", you type "Italian A1" in the search bar and are then teleported to it instantly)...

**Comment by valiantone** [comment]
> Could you please clarify what you mean exactly by "gameplan"? Do you mean a possible work plan to design and complete the product with other UX techies?

Okay @vanniberna , so the game plan would be a list of features that you can imagine for each of the three tracks: WebDev, UX and Data Science. You kind of already added them in a way in the previous comments. With the exception of UX, over there, maybe you could list out as a set of small tasks, for ex. create initial survey to understand user needs; synthesize results of survey to formalize user needs etc.

Once this is done, I would say, start promoting your project and adding potentially interested techies! And offcourse, keep us in the loop if you hit any road-blocks

**Comment by vanniberna** [comment]
> > Could you please clarify what you mean exactly by "gameplan"? Do you mean a possible work plan to design and complete the product with other UX techies?
> 
> Okay @vanniberna , so the game plan would be a list of features that you can imagine for each of the three tracks: WebDev, UX and Data Science. You kind of already added them in a way in the previous comments. With the exception of UX, over there, maybe you could list out as a set of small tasks, for ex. create initial survey to understand user needs; synthesize results of survey to formalize user needs etc.
> 
> Once this is done, I would say, start promoting your project and adding potentially interested techies! And offcourse, keep us in the loop if you hit any road-blocks

Hey @valiantone , thanks for your reply! 

Great, so as far as the game plan for other assisting UX techies goes, I would suggest the following steps:
- assist in the ideation process;
- help to generate wireframes for the project's UI (responding to the principles I listed in my original application);
- turn these wireframes into a fully-fledged prototype;
- help to find formats and problem scenarios in which to test the prototype with users (students, course instructors). These don't have to materialize out of thin air, of course — they would mainly stem from a university where I currently work as an instructor myself;
- respond to users' feedback to refine the prototype.

**Comment by valiantone** [comment]
Amazing! Great work, love that you are so pro-active! Now go a ahead and find your team of DS and WD contributors. Here is the minimum acceptance criteria for team formation @vanniberna :
- [x] 2DS
- [x] 2WD **(1BE & 1FE)**
- [x] 2UX

Assign them to this issue if they agree!
